ucp/useraccounts_module.php

```diff
@@ -308,8 +258,8 @@ function main($id, $mode)
 			case 'select_badges':
 			{
 				// Set up vars for mode.
-				$markedbadges = $this->request->variable('marked_badges', array(0)); // badge_ids
-				$submitmarked = $this->request->is_set_post('marked_badges') || empty($markedbadges) && $action == 'submit_marked' ? true : false;
+				$markedbadges = (array)$this->request->variable('marked_badges', array(0)); // badge_ids
```

Comment:
`array(0)` is an array containing the number `0`.  Shouldn't this be an empty array (`array()`)?

Reply:
[array(0) means array has 0 index (empty)](https://stackoverflow.com/questions/13232737/var-dump-array0-php#comment18025524_13232737), with `array()` ti does not collect the values from `name="markedkeys[]" value=<int>`` in the template. phpBB uses this wherever it requests values from marked checkboxes and in a lot of other places too.